llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2224,12 +2224,17 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
   // (~X) - (~Y) --> Y - X
   // This is placed after the other reassociations and explicitly excludes a
   // sub-of-sub pattern to avoid infinite looping.
```

Comment:
The comment above is outdated / redundant with the one below.

Reply:
Well, at least the part about the "sub-of-sub pattern".

Reply:
Although the whole thing is deprecated as ensuring we consume a `not` is what precludes inf loop.